scripts/security_helpers.py

```
@@ -58,3 +65,135 @@ def normalize_https_url(
     if strip_query:
         sanitized = sanitized._replace(query="")
     return urlunparse(sanitized)
+
+
+def validate_slug(value: str, *, field_name: str) -> str:
+    slug = (value or "").strip()
+    if not slug or not _SLUG_RE.fullmatch(slug):
+        raise ValueError(f"Invalid {field_name}: {value!r}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)